File: src/data_types/testdataset.py
"""
:Author: Paul Erlenwein
:Since: 2020/11/20

This is the module containing the TestDataSet datatype.
"""

from dataclasses import dataclass
from uuid import uuid4

from data_types.evaluation import Evaluation


@dataclass
class TestDataSet:
    """This is a ``TestDataSet``. It is the representation
    of a comparison of a **manEval** and an **autoEval**.
    """

    # Identifying Attributes

    # TODO: Add trivial attributes like Exercise name, ... for FE representation?

    expert_model_id: str
    student_model_id: str
    meta_model_type: str
    mcs_identifier: str
    mcs_version: str

    # Evaluations
    auto_eval_id: str = None
    man_eval_id: str = None

    auto_eval: Evaluation = None
    man_eval: Evaluation = None

File: frontend/src/app/interfaces/test-data-set.ts
import { Evaluation } from './evaluation';

export interface TestDataSet {
  expert_model_id: string;
  student_model_id: string;
  meta_model_type: string;
  mcs_identifier: string;
  mcs_version: string;
  auto_eval_id: string;
  man_eval_id: string;
  auto_eval: Evaluation
  man_eval: Evaluation
}
